### ISSUE TITLE
Adding Testing capabilities with Tendermint via Command Line

### ISSUE THREAD

**Issue by colin-axner** [FourthState/plasma-mvp-sidechain#35]
Before moving on and adding more complexity, we want to ensure that our current base layer of our blockchain works as expected. 

At it's current state on master it should be capable of:
- Sending transactions
- Updating State
- Validating transactions
- Rejecting invalid transactions

To sync up with Tendermint via Command Line we will need:
Plasma Node 
Plasma Light Client

The Plasma node is responsible for maintaining the application and processing received transactions. I have already done this in my branch colin/client. The node is simply just using the sdk's server client to start up a node with the plasma child chain. The node will be running a tendermint node also. To test out this functionality simply, pull my branch, do go build in plasmad/ directory, then do go install. Run `plasmad init` and then` plasmad start`. You should see empty blocks being committed by the tendermint node. 

The Plasma Light Client is responsible for doing any users functions such as sending a transaction or querying their utxo's. The light client will need to be in communication with the node in order to do this. 

I was planning on building the client directory as follows:
```
client/
    plasmad/
        main.go
    plasmacli/
        main.go
        cmd/
            sendtx.go
            version.go
            more.go
            commands.go
            here.go
```

**Comment by colin-axner** [comment]
Expanding further on plasmacli:

The light client will need to have some sort of key management system in order to sign off on transactions and then broadcast them. Cosmos-SDK currently uses the go-crypto [keybase](https://github.com/tendermint/go-crypto/blob/master/keys/types.go) to manage these keys and maps the name of a key in string to the actual key. This is nice UI as users simply need to give their addresses names and specify those names via flags to do any commands with them.

Since we need to use geth-crypto to stay compatible with Ethereum we will need to use a different type of key management system. Go-Ethereum has a [keystore](https://github.com/ethereum/go-ethereum/blob/master/accounts/keystore/keystore.go) which implements their BackEnd interface which is described in the [docs](https://godoc.org/github.com/ethereum/go-ethereum/accounts) for their accounts package. The keystore can generate new account keys and stores the accounts which can then sign off on transactions. I believe the [find](https://github.com/ethereum/go-ethereum/blob/master/accounts/keystore/keystore.go#L368) function can retrieve all the info necessary to sign off on transaction by providing the account address (An account struct is simply the address with an optional URL field).

We can use a custom context similar to [this](https://github.com/cosmos/cosmos-sdk/blob/master/client/context/types.go) so the account address to sign a transaction for is passed in via flags corresponding to the sendtx command. Is there any way we could also add in the UI feature of referencing certain addresses by a string rather than the full address itself? 

If I'm not mistaken or missing something, I believe this construction can work. I'm still working out the details for storing/accessing the keystore. 

Thoughts?

**Comment by AdityaSripal** [comment]
Current idea: Implement keybase interface in go-crypto to handle Ethereum sigs. Most methods can be taken from standard implementation, except for methods that are signature-scheme specific.

**Comment by AdityaSripal** [comment]
Closed in #38
